### PR TITLE
Run Zeebe as non-root user in docker container

### LINKED ISSUE #3584: Docker Containers shouldn't run as Root
**Describe the bug**
Today, our docker containers are running as root user.
This might cause problems in some platforms and in general it is a bad practice.

**To Reproduce**
Steps to reproduce the behavior

If possible add a minimal reproducer code sample
Reported by a community user using the Zeebe Helm Charts. Anyone else running in a managed environment will face the same problem, no matter if they are using helm or manually deploying our containers. 
https://github.com/zeebe-io/zeebe-cluster-helm/pull/7

**Expected behavior**

Run all containers as non root users

**Environment:**
 - OS: Kubernetes
 - Zeebe Version: [e.g. 0.22.0-alpha2]
 - Configuration: HELM charts

### PR DESCRIPTION
**Pull request by menski**
## Description

Adjusts the Zeebe docker image to run Zeebe as non-root `zeebe` user 

## Related issues

closes #3584

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

**Comment by npepinpe** [comment]
If we run as a non root user, this will prevent us from adding packages at runtime, correct? So we won't be able to install diagnostic tools?

**Comment by menski** [comment]
You can always execute a command inside the container using a different user like `root`

**Comment by menski** [comment]
I create an image with this change for Zeebe 0.23.1 and pushed it to docker hub under `camunda/zeebe:0.23.1-non-root`. Let's test it in our environments before we merge this.

- [ ] @npepinpe please check if we can still use it in our benchmark setup
- [ ] @menski please check Camunda cloud
- [ ] @pihme please check if it still works in Windows
- [ ] @salaboy please check if it works with helm and maybe environments where `root` was a problem

Thanks everyone for helping

**Comment by menski** [comment]
It seems the container breaks the upgrade tests, I will investigate this.

File: Dockerfile [hunk 1]
```diff
@@ -1,30 +1,47 @@
...
+    tar xfvz ${TMP_ARCHIVE} --strip 1 -C ${TMP_DIR} && \
+    # already create volume dir to later have correct rights
+    mkdir ${TMP_DIR}/data
+
+ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini ${TMP_DIR}/bin/tini
+COPY docker/utils/startup.sh ${TMP_DIR}/bin/startup.sh
+RUN chmod +x -R ${TMP_DIR}/bin/
+
+## Final Image
+FROM openjdk:11-jre
```
Comment (npepinpe) [inline Dockerfile:22]: If we're switching the pipeline any way, might as well go for `openjdk:11-jre-slim`? Then you don't need to purge apt repos and so on

Comment (menski) [reply Dockerfile:22]: I will give it a try

**Comment by menski** [comment]
The problem seems to be that if you mount a volume to the `data/` directory from the outside, i.e. not managed from docker itself, then the permissions are not correct and the broker fails to read/write the partitions etc.

```
docker run --rm -v ./data:/usr/local/zeebe/data camunda/zeebe:0.23.1-non-root
```

```
2020-05-05 11:41:12.013 [] [main] ERROR org.springframework.boot.SpringApplication - Application run failed                                                                      
java.lang.IllegalStateException: Failed to execute CommandLineRunner                                                                                                             
        at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:787) [spring-boot-2.2.5.RELEASE.jar:2.2.5.RELEASE]                                       
        at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:768) [spring-boot-2.2.5.RELEASE.jar:2.2.5.RELEASE]                                      
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:322) [spring-boot-2.2.5.RELEASE.jar:2.2.5.RELEASE]                                              
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1226) [spring-boot-2.2.5.RELEASE.jar:2.2.5.RELEASE]                                             
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1215) [spring-boot-2.2.5.RELEASE.jar:2.2.5.RELEASE]                                             
        at io.zeebe.broker.StandaloneBroker.main(StandaloneBroker.java:46) [zeebe-distribution-0.23.1.jar:0.23.1]                                                                
Caused by: io.zeebe.util.exception.UncheckedExecutionException: Failed to start broker                                                                                           
        at io.zeebe.broker.Broker.internalStart(Broker.java:142) ~[zeebe-broker-0.23.1.jar:0.23.1]                                                                               
        at io.zeebe.util.LogUtil.doWithMDC(LogUtil.java:21) ~[zeebe-util-0.23.1.jar:0.23.1]                                                                                      
        at io.zeebe.broker.Broker.start(Broker.java:115) ~[zeebe-broker-0.23.1.jar:0.23.1]                                                                                       
        at io.zeebe.broker.StandaloneBroker.run(StandaloneBroker.java:59) ~[zeebe-distribution-0.23.1.jar:0.23.1]                                                                
        at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:784) ~[spring-boot-2.2.5.RELEASE.jar:2.2.5.RELEASE]                                      
        ... 5 more                                                                                                                                                               
Caused by: java.lang.IllegalArgumentException: could not create Raft system directory directory: /usr/local/zeebe/data/system                                                    
        at org.agrona.IoUtil.ensureDirectoryExists(IoUtil.java:241) ~[agrona-1.4.1.jar:1.4.1]                                                                                    
        at io.zeebe.broker.clustering.atomix.AtomixFactory.fromConfiguration(AtomixFactory.java:76) ~[zeebe-broker-0.23.1.jar:0.23.1]                                            
        at io.zeebe.broker.Broker.atomixCreateStep(Broker.java:196) ~[zeebe-broker-0.23.1.jar:0.23.1]                                                                            
        at io.zeebe.broker.Broker.lambda$initStart$0(Broker.java:162) ~[zeebe-broker-0.23.1.jar:0.23.1]                                                                          
        at io.zeebe.broker.bootstrap.StartProcess.lambda$startStepByStep$2(StartProcess.java:60) ~[zeebe-broker-0.23.1.jar:0.23.1]                                               
        at io.zeebe.broker.bootstrap.StartProcess.takeDuration(StartProcess.java:88) ~[zeebe-broker-0.23.1.jar:0.23.1]                                                           
        at io.zeebe.broker.bootstrap.StartProcess.startStepByStep(StartProcess.java:58) ~[zeebe-broker-0.23.1.jar:0.23.1]                                                        
        at io.zeebe.broker.bootstrap.StartProcess.takeDuration(StartProcess.java:88) ~[zeebe-broker-0.23.1.jar:0.23.1]                                                           
        at io.zeebe.broker.bootstrap.StartProcess.start(StartProcess.java:43) ~[zeebe-broker-0.23.1.jar:0.23.1]                                                                  
        at io.zeebe.broker.Broker.internalStart(Broker.java:135) ~[zeebe-broker-0.23.1.jar:0.23.1]                                                                               
        at io.zeebe.util.LogUtil.doWithMDC(LogUtil.java:21) ~[zeebe-util-0.23.1.jar:0.23.1]                                                                                      
        at io.zeebe.broker.Broker.start(Broker.java:115) ~[zeebe-broker-0.23.1.jar:0.23.1]                                                                                       
        at io.zeebe.broker.StandaloneBroker.run(StandaloneBroker.java:59) ~[zeebe-distribution-0.23.1.jar:0.23.1]                                                                
        at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:784) ~[spring-boot-2.2.5.RELEASE.jar:2.2.5.RELEASE]                                      
        ... 5 more 
```

**Comment by npepinpe** [comment]
Ping me if you need any help there :+1: or just need a rubber :duck:

**Comment by menski** [comment]
@pihme and @salaboy I removed you for review as the container is not working with data volumes mounted. I have to think how to fix it before I will ask you again for a review. thanks

**Comment by npepinpe** [comment]
Seems to work, but throughput was way down compared to our baseline :thinking: I don't think this would be a problem from something you added so take it with a grain of salt, I'll look more into it

**Comment by menski** [comment]
@npepinpe Was the container actually able to start in our benchmark? Based on my research I expected it to fail to create the partition directories or read the configuration.

**Comment by npepinpe** [comment]
Yes, it could process albeit throughput was down from ~160 to ~30, I need to investigate more

**Comment by npepinpe** [comment]
Scratch that, I need to re run again, I ram it from the wrong setup so I wasn't checking your image, my bad

**Comment by menski** [comment]
replaced by #4558 and #4541

the only thing missing is the testing added in this PR. we might want to revisit that later.

**Comment by npepinpe** [comment]
There's a few more things but I will do those separately - introducing the docker profile (which removes a good 30mb by fixing variants), and copying dependencies directly as opposed to a tar ball (so third party deps are one layer, and our code is another much smaller layer).

**Pull request closed without merging**